src/vs/server/node/remoteTerminalChannel.ts

```diff
@@ -107,72 +107,72 @@ export class RemoteTerminalChannel extends Disposable implements IServerChannel<
 
 	async call(ctx: RemoteAgentConnectionContext, command: string, args?: any): Promise<any> {
 		switch (command) {
-			case '$restartPtyHost': return this._ptyService.restartPtyHost?.apply(this._ptyService, args);
+			case '$restartPtyHost': return this._ptyHostService.restartPtyHost.apply(this._ptyHostService, args);
 
 			case '$createProcess': {
 				const uriTransformer = createURITransformer(ctx.remoteAuthority);
 				return this._createProcess(uriTransformer, <ICreateTerminalProcessArguments>args);
 			}
-			case '$attachToProcess': return this._ptyService.attachToProcess.apply(this._ptyService, args);
-			case '$detachFromProcess': return this._ptyService.detachFromProcess.apply(this._ptyService, args);
+			case '$attachToProcess': return this._ptyHostService.attachToProcess.apply(this._ptyHostService, args);
+			case '$detachFromProcess': return this._ptyHostService.detachFromProcess.apply(this._ptyHostService, args);
 
-			case '$listProcesses': return this._ptyService.listProcesses.apply(this._ptyService, args);
-			case '$getPerformanceMarks': return this._ptyService.getPerformanceMarks.apply(this._ptyService, args);
-			case '$orphanQuestionReply': return this._ptyService.orphanQuestionReply.apply(this._ptyService, args);
-			case '$acceptPtyHostResolvedVariables': return this._ptyService.acceptPtyHostResolvedVariables?.apply(this._ptyService, args);
+			case '$listProcesses': return this._ptyHostService.listProcesses.apply(this._ptyHostService, args);
+			case '$getPerformanceMarks': return this._ptyHostService.getPerformanceMarks.apply(this._ptyHostService, args);
+			case '$orphanQuestionReply': return this._ptyHostService.orphanQuestionReply.apply(this._ptyHostService, args);
+			case '$acceptPtyHostResolvedVariables': return this._ptyHostService.acceptPtyHostResolvedVariables.apply(this._ptyHostService, args);
 
-			case '$start': return this._ptyService.start.apply(this._ptyService, args);
-			case '$input': return this._ptyService.input.apply(this._ptyService, args);
-			case '$acknowledgeDataEvent': return this._ptyService.acknowledgeDataEvent.apply(this._ptyService, args);
-			case '$shutdown': return this._ptyService.shutdown.apply(this._ptyService, args);
-			case '$resize': return this._ptyService.resize.apply(this._ptyService, args);
-			case '$clearBuffer': return this._ptyService.clearBuffer.apply(this._ptyService, args);
-			case '$getInitialCwd': return this._ptyService.getInitialCwd.apply(this._ptyService, args);
-			case '$getCwd': return this._ptyService.getCwd.apply(this._ptyService, args);
+			case '$start': return this._ptyHostService.start.apply(this._ptyHostService, args);
```

Comment:
I still find this syntax a bit confusing - passing in the `ptyHostService` to it's function application... 🤔

Reply:
That's binding the `this` variable inside the function, which you need to do when you use apply, otherwise it would be `undefined`.